### ISSUE TITLE
zen_draw_pulldown_products improvement

### ISSUE THREAD

**Issue by proseLA** [zencart/zencart#4096]
(minor aside... i like the new start of an issue.  :thumbsup:)

i have no strong feeling about the new naming on this function; so whatever is decided, i like.

problem:
adding a product to a special or featured page.  there are 2 options, to add by selection or to add by id.  if one chooses to add by selection, the whole listing of the store's products comes up.  very difficult to find the problem you want.

solution:
- add a keywords parameter to the `zen_draw_pulldown_products` function that makes use of the new in v158, `zen_build_keyword_where_clause` function.  this parameter would be the last parm in the function and have a default of empty.
- we could then add a new form to allow for searching and limit what shows up on the selection menu.
- while i do not use attributes nor any pulldowns associated with it, i can only see the potential benefit there as well.

here are 2 screenshots as proof of concept (using my own specialized test data), that hopefully illustrates the solution.

![Screenshot from 2020-11-19 10-14-44](https://user-images.githubusercontent.com/1095136/99706827-7f836580-2a50-11eb-83d0-7b13333d0ac9.png)
![Screenshot from 2020-11-19 10-15-20](https://user-images.githubusercontent.com/1095136/99706840-83af8300-2a50-11eb-8d81-a0534e39f3ca.png)

i am a fan of consistent searching and fan (as well as author) of  the `zen_build_keyword_where_clause` function.  i think its use in the building of pulldowns would be a big improvement in this area.

i am also willing to do a PR on it if approved and once the other PR(s) with this function(s) get sorted out.

best.

**Comment by torvista** [comment]
I would agree that a search option would be an good enhancement for these pages.
Although I have 6k+ products and the dropdown takes less than a couple of seconds to appear. But since I use SKUs, finding products is very easy.

An instant search where results appear after a few characters have been entered, would be something I would like to see implemented for all search fields, since we are in 2021!

**Comment by proseLA** [comment]
i've looked at that instant search you have and could not get it to work to my liking.  the styling was off to the point that i could not see where the products were; and unfortunately my time to debug that is limited.  my css skills are also a bit lacking...

this proposed change builds on existing infrastructure.  and if the change to the zc function is accepted, the code changes are minimal.

again, this is a change that i think would benefit the community, not just one individual.  assuming a store owner does NOT know the model number and has 6k+ products, scrolling through all of them to find one product, i think is a real pain.  if the store owner knew that it came from XYZ manufacturer, and all of that manufacturers model numbers start with XYZ, and the product was a fire table, one could search for `xyz fire table` and get a list of product matches.  and the keywords do not have to be in order.  `table xzy fire` would pull up the same product list.

i agree that instant search could be a valuable addition, but well outside the scope of what i am proposing.

**Comment by torvista** [comment]
> i've looked at that instant search you have and could not get it to work to my liking

Ooh, I can't imagine what state that is in. I remember struggling to understand the wonky workings of the original and butchered it enough for me to use. Then I think I put it up on GitHub intending  to review it, and never did, so apologies if it is half-assed.
I was only suggesting the _concept_ of an instant search, not that one!

We actually live and die by the model numbers, not the product names, as they are unique, and the product names too wordy. Still, a search would be fine.
.
